internal/monitor/diskmonitor.go

```diff
@@ -254,50 +265,63 @@ func (monitor *DiskMonitor) readZfsDatasets() map[string]ZfsDataset {
-		pathToQuery = filepath.Join(hostFsPrefix, pathToQuery)
-	}
-
-	tempMap := make(map[string]uint64)
-
-	temps, err := ini.Load(pathToQuery)
-	if err != nil {
-		slog.Error("Disk unable to read disks.ini", "error", slog.String("error", err.Error()))
-	}
-
-	for _, section := range temps.Sections() {
-		if section.Name() == "DEFAULT" {
-			continue
-		}
-
-		sectionName := strings.Trim(section.Name(), "\"")
-
-		slog.Debug("Disk reading section", "section", sectionName)
-		var temp uint64
-		tempString, err := section.GetKey("temp")
-		if err != nil {
-			slog.Error("Disk unable to read disks.ini section", "section", sectionName, "error", slog.String("error", err.Error()))
-			continue
-		}
-
-		if tempString.String() != "*" {
-			temp, err = strconv.ParseUint(tempString.String(), 10, 16)
-			if err != nil {
-				slog.Error("Disk unable to parse disk temp", "string", tempString.String(), "error", slog.String("error", err.Error()))
-			}
-		} else {
-			slog.Debug("Disk temp unavailable", "disk", sectionName)
-		}
-
-		tempMap[sectionName] = temp
-	}
-
-	return tempMap
+func readDiskIni() map[string]DiskIni {
+    pathToQuery := "/var/local/emhttp/disks.ini"
+
+    var hostFsPrefix, isSet = os.LookupEnv("HOSTFS_PREFIX")
+    if isSet {
+        slog.Debug("Disk host prefix is set", "value", hostFsPrefix)
+        pathToQuery = filepath.Join(hostFsPrefix, pathToQuery)
+    }
+
+    diskIniMap := make(map[string]DiskIni)
+
+    disks, err := ini.Load(pathToQuery)
+    if err != nil {
+        slog.Error("Disk unable to read disks.ini", "error", slog.String("error", err.Error()))
+    }
+
+    for _, section := range disks.Sections() {
+        if section.Name() == "DEFAULT" {
+            continue
+        }
+
+        sectionName := strings.Trim(section.Name(), "\"")
+
+        slog.Debug("Disk reading section", "section", sectionName)
+
+        idString, err := section.GetKey("id")
+        if err != nil {
+            slog.Error("Disk unable to read disks.ini section", "section", sectionName, "error", slog.String("error", err.Error()))
+            continue
+        }
+
+        var temp uint64
+        tempString, err := section.GetKey("temp")
+        if err != nil {
+            slog.Error("Disk unable to read disks.ini section", "section", sectionName, "error", slog.String("error", err.Error()))
+            continue
+        }
+
+        if tempString.String() != "*" {
+            temp, err = strconv.ParseUint(tempString.String(), 10, 16)
+            if err != nil {
+                slog.Error("Disk unable to parse disk temp", "string", tempString.String(), "error", slog.String("error", err.Error()))
+            }
+        } else {
+            slog.Debug("Disk temp unavailable", "disk", sectionName)
+        }
+
+		spunDown, err := section.GetKey("spundown")
+
+        diskIniMap[sectionName] = DiskIni{
+            ID:   idString.String(),
+            Temp: temp,
+			Spundown: spunDown.String() == "1",
+        }
```

Comment:
weird indentation here

Reply:
Yeah I guess because I just tabs. Will change to spaces